### ISSUE TITLE
Generic typing of create namespace ops is broken

### ISSUE THREAD

**Issue by ctrueden** [imagej/imagej-ops#160]
The create namespace has several ops that use generics in unsafe and invalid ways. Whenever possible, it is best for methods which returns objects with generic parameters to accept method arguments that also specify those types. Otherwise, the compiler has a hard time guaranteeing that those generic type parameters will be fulfilled as expected.

**Comment by ctrueden** [comment]
Regarding `DefaultCreateImg`: we should split it into multiple ops with different signatures, since you need to pass either a `T type` or an `ImgFactory<T>` or a `Class<T>`—_something_ to provide the `T` at runtime. And the cast from `dims` to `Img<?>` to (effectively) `Img<T>` is also bad—we should just have a separate (higher priority) op implementation that takes an `Img<T>` or `RandomAccessible<T>` or whatever typed on `T` and thus will have an associated `ImgFactory<T>` and not need to do a potentially broken cast that way. And then `DefaultCreateImg` can accept none of the `T` parameters at all and simply be typed on `DoubleType` directly, since that's ultimately what it will produce.

**Comment by dietzc** [comment]
@ctrueden  https://github.com/imagej/imagej-ops/pull/194 does this help?

**Comment by ctrueden** [comment]
@dietzc Yes I saw that, and it is exactly what we need to do! We should of course review and merge that PR. Then we can review the state of the create namespace as a whole again. If all the generics issues are resolved, we then close this issue—and if not, we enumerate the remaining problems here and solve them.

**Comment by ctrueden** [comment]
The current plan is to use `FunctionViaComputer` in the `net.imagej.ops.chain` package to separate the computer and function portion of ops that use generics in their output type.

See also the [remove-varargs](https://github.com/imagej/imagej-ops/compare/remove-varargs) branch where this work is likely to proceed.

**Comment by ctrueden** [comment]
Big progress in 267e2655191cf10da6589ba9cd3864db664808b1. But I am going to leave this issue open until I have time to verify that _all_ `create` ops have been fixed.

**Comment by dietzc** [comment]
> Big progress in 267e265.

:+1:
